services/explorer/src/pages/BlockData/DAGDataContainer.effects.tsx

```diff
@@ -0,0 +1,45 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useQuery } from 'urql';
+import { dagDetailsQuery } from '../../api';
+import { useExplorerNetwork } from '../../hooks/useExplorerNetwork';
+import { useExplorerLoader } from '../../hooks/useLoader';
+import { DagBlock, Transaction } from '../../models';
+
+export const useDAGDataContainerEffects = (hash: string) => {
+  const { currentNetwork } = useExplorerNetwork();
+  const [blockData, setBlockData] = useState<DagBlock>({} as DagBlock);
+  const [transactions, setTransactions] = useState<Transaction[]>([
+    {} as Transaction,
+  ]);
+  const [{ fetching, data }] = useQuery({
+    query: dagDetailsQuery,
+    variables: {
+      hash,
+    },
+    pause: !hash,
+  });
+  const { initLoading, finishLoading } = useExplorerLoader();
+
+  const fetchBlockDetails = useCallback(() => {
+    setTimeout(() => {
+      console.log('dddd: ', data);
+      setBlockData(data?.dagBlock);
```

Comment:
This should be inside an if statement if there is data?.dagBlock - sometimes it comes as null